src/txtai_mcp_server/tools/search.py

```
@@ -112,7 +112,14 @@ async def semantic_search(
                         score = result["score"]
                         
                         # Try to get the document text from the cache
-                        text = get_document_from_cache(doc_id) or "No text available"
+                        text = get_document_from_cache(doc_id)
+                        if not text:
+                            try:
+                                app = get_txtai_app()
+                                sql_result = app.search(f"select text from txtai where id = '{doc_id}'")
```

Ruff (0.15.2)

[error] Possible SQL injection vector through string-based query construction

(S608)